MoTrPAC/motrpac-frontend
# Connects to #102. Auth0 redirecting non-authorized users to error page.

### Key Changes

1. Created a new rule in **Auth0** to handle Auth0-authenticated users who have not been authorized to access data.
2. Refactored `ErrorPage` component in handling non-authorized users redirect.
3. Refactored `Callback` component in handling non-authorized users redirect.
4. Refactored `PrivateRoute` component in handling non-authorized users redirect.
5. Refactored `Dashboard` component in handling non-authorized users redirect.
6. Patch updated a number of dependency modules.

### Technical Notes

1. Requires `node.js v14.15.4` to build the app and run the tests.
2. Running `rm -fr node_modules/` and then `yarn install` prior to running `yarn inspect-branch` is recommended.
3. If needed, replace the dummy `src/data/pass1b_06_pheno_*.json` files with the ones downloaded from GCS bucket to view the rendered forest plots.

### Steps to test

1. Edit the **Auth0 Rule** named _Redirect Non-authorized Users_ by replacing the `motrpac-data.org` with `localhost:3000` in the `logoutUrl` variable. Save the changes.
2. Try to sign into the app with a user account that has no access permission to the MoTrPAC data.
3. Expect to be redirected to the `^/error` page whose previously implemented **Return Home** button is no longer present.
4. Try to sign into the app with a user account that has been granted access to data.
5. Expect to be taken to the `^/dashboard` page.
